lms/djangoapps/course_api/blocks/transformers/extra_fields.py

```diff
@@ -0,0 +1,69 @@
+"""
+Extra Fields Transformer
+"""
+from django.conf import settings
+
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+
+
+class ExtraFieldsTransformer(BlockStructureTransformer):
+    """
+    Shows ExtraFields specified in COURSE_BLOCKS_API_EXTRA_FIELDS LMS Django Settings variable
+    """
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+
+    @classmethod
+    def name(cls):
+        return "blocks_api:extra_fields"
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this transformer's
+        transform method.
+        """
+        requested_field_names = cls.get_requested_extra_fields()
+
+        block_structure.request_xblock_fields('category', *requested_field_names)
+
+    @classmethod
+    def get_requested_extra_fields(cls):
+        """
+        Returns the names of the requested extra fields
+        """
+        try:
+            return [field_name for block_type, field_name in settings.COURSE_BLOCKS_API_EXTRA_FIELDS]
+        except AttributeError:
+            return []
+
+    def transform(self, usage_info, block_structure):
+        """
+        Mutates block_structure based on the given usage_info.
+        """
+        if len(self.get_requested_extra_fields()) == 0:
+            return
+
+        requested_fields = [dict(
+            block_type=block_type,
+            field_name=field_name,
+        ) for block_type, field_name in settings.COURSE_BLOCKS_API_EXTRA_FIELDS]
```

Comment:
Nitpick: This conversion to a `dict` doesn't really do much that I can see. I think this would be cleaner if you just remove these four lines and replace the inner loop below,

```python
for requested_field in requested_fields:
    ... requested_field['block_type'] ... requested_field['field_name']
```

with 

```python
for requested_block_type, requested_field_name in settings.COURSE_BLOCKS_API_EXTRA_FIELDS:
    ... requested_block_type ... requested_field_name
```